test/extensions/filters/network/redis_proxy/config_test.cc

```diff
@@ -66,63 +66,6 @@ settings: {}
                           ProtoValidationException, "embedded message failed validation");
 }
 
-TEST(RedisProxyFilterConfigFactoryTest,
-     DEPRECATED_FEATURE_TEST(RedisProxyCorrectProtoLegacyCluster)) {
-  TestDeprecatedV2Api _deprecated_v2_api;
-  Runtime::LoaderSingleton::getExisting()->mergeValues(
-      {{"envoy.deprecated_features:envoy.config.filter.network.redis_proxy.v2.RedisProxy.cluster",
-        "true"},
-       {"envoy.deprecated_features:envoy.extensions.filters.network.redis_proxy.v3.RedisProxy."
-        "hidden_envoy_deprecated_cluster",
-        "true"}});
-
-  const std::string yaml = R"EOF(
-cluster: fake_cluster
-stat_prefix: foo
-settings:
-  op_timeout: 0.02s
-  )EOF";
-
-  envoy::extensions::filters::network::redis_proxy::v3::RedisProxy proto_config{};
-  TestUtility::loadFromYamlAndValidate(yaml, proto_config, true, false);
-  NiceMock<Server::Configuration::MockFactoryContext> context;
-  RedisProxyFilterConfigFactory factory;
-  Network::FilterFactoryCb cb = factory.createFilterFactoryFromProto(proto_config, context);
-  EXPECT_TRUE(factory.isTerminalFilterByProto(proto_config, context));
-  Network::MockConnection connection;
-  EXPECT_CALL(connection, addReadFilter(_));
-  cb(connection);
-}
-
-TEST(RedisProxyFilterConfigFactoryTest,
```

Comment:
In general it's safe to delete tests only as long as there is coverage for the new fields that replace them (this isn't always the case). In this case, looks like `catch_all_route` is the new way to do it and there are tests for that. So, LGTM.